### PR TITLE
fix(duplicated-query): add cost query name validation

### PR DESCRIPTION
**Pull request by piggggggggy**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [x] Bug fixes
- [ ] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [ ] mirinae
  - [ ] etc 

- [x] Apps
  - [ ] storybook
  - [x] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Description
SSIA

image

### CODE REVIEW

**Review by skdud4659 — APPROVED** [review]
LGTM!

File: apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisQueryFormModal.vue [hunk 1]
```diff
@@ -106,6 +118,22 @@ watch(() => state.proxyVisible, (visible) => {
...
+                filter: [{ k: 'user_id', v: store.state.user.userId, o: 'eq' }],
+                only: ['name'],
+            },
+        });
+        if (status === 'succeed' && response?.results) {
+            state.existingCostQuerySetNameList = [...state.existingCostQuerySetNameList, ...response.results.map((query) => query.name)];
+        }
+    } catch (e) {
+        state.existingCostQuerySetNameList = [...state.existingCostQuerySetNameList];
+    }
```
Comment (WANZARGEN) [inline apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisQueryFormModal.vue:135]: issue: I think ErrorHandler is needed.

File: apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisQueryFormModal.vue [hunk 2]
```diff
@@ -106,6 +118,22 @@ watch(() => state.proxyVisible, (visible) => {
...
+onMounted(async () => {
+    try {
+        const { status, response } = await costQuerySetFetcher({
+            query: {
+                filter: [{ k: 'user_id', v: store.state.user.userId, o: 'eq' }],
+                only: ['name'],
+            },
+        });
+        if (status === 'succeed' && response?.results) {
+            state.existingCostQuerySetNameList = [...state.existingCostQuerySetNameList, ...response.results.map((query) => query.name)];
```
Comment (WANZARGEN) [inline apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisQueryFormModal.vue:131]: issue: What if the modal is popped up several times? Then previous existingCostQuerySetNameList will be accumulated.
We don't know that existingCostQuerySetNameList is initiated one or used one.
I think it's hard to debug if the code works differently depends on the context.
I recommend to separate the managedCostQuerySetIdList and existingCostQuerySetNameList, and use merged list for check.

**Review by WANZARGEN — COMMENTED** [review]
Thank you for hard work! I left some comments, so please have a look 🙏

File: apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisQueryFormModal.vue (hunk 2 above)
Comment (piggggggggy) [reply apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisQueryFormModal.vue]: Thank for catching issue that I missed. I rewrite that code. Check plz!

File: apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisQueryFormModal.vue [hunk 3]
```diff
@@ -128,11 +130,11 @@ onMounted(async () => {
             },
         });
         if (status === 'succeed' && response?.results) {
-            state.existingCostQuerySetNameList = [...managedCostQuerySetIdList, ...response.results.map((query) => query.name)];
+            state.existingCostQuerySetNameList = [...response.results.map((query) => query.name)];
```
Comment (WANZARGEN) [inline apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisQueryFormModal.vue:133]:
```suggestion
state.existingCostQuerySetNameList = response.results.map((query) => query.name);
```

**Review by WANZARGEN — APPROVED** [review]
Thank you for applying my comments! I left one more suggestion! 😄

**Pull request merged**